programs/survival/tputils.sc

```diff
@@ -0,0 +1,283 @@
+global_max_homes = 20; //Edit to allow more/less homes
+global_operator_permission_level = 2;
+
+__config()->{
+    'scope' -> 'global',
+    'commands' ->
+    {   
+        'home add <nameh>' -> '_hadd',
+        'home tp <hlocation>' -> '_htp',
+        'home remove <hlocation>' -> '_hremove',
+        'home list' -> '_hlist',
+        'warp add <namew> <description>' -> '_wadd',
+        'warp tp <wlocation>' -> '_wtp',
+        'warp remove <wlocation>' -> '_wremove',
+        'warp menu' -> '_wmenu'
+    },
+    'arguments' ->
+    {
+        'hlocation' -> {
+            'type' -> 'text',
+            'suggester' -> _(args) -> (
+                keys(global_hdata:str(player()~'uuid'))
+            ),
+            'case_sensitive' -> false
+        },
+
+        'nameh' -> {
+            'type' -> 'text',
+            'suggest' -> ['Spawn','Base','End Portal','Nether Portal'],
+            'case_sensitive' -> false
+        },
+
+        'namew' -> {
+            'type' -> 'string',
+            'suggest' -> ['Spawn','Marketplace','Arena'],
+            'case_sensitive' -> false
+        },
+
+        'description' -> {
+            'type' -> 'text'
+        },
+
+        'wlocation' -> {
+            'type' -> 'text',
+            'suggester' -> _(args) -> (
+                keys(global_wdata)
+            ),
+            'case_sensitive' -> false
+        }
+    }
+};
+
+
+//Initializing homes and warps data
+
+if(read_file('hdata','JSON') == null,
+    global_hdata = {},
+    //else when file already exists
+    global_hdata = read_file('hdata','JSON');;
+);
+if(read_file('wdata','JSON') == null,
+    global_wdata = {},
+    //else when file already exists
+    global_wdata = read_file('wdata','JSON');;
+);
+
+
+
+//Home functions
+
+_hadd(name) -> (
+    
+    player=player();
+    uuid = str(player~'uuid');
+
+    if(length(global_hdata:uuid)>global_max_homes,
+        print(player,format('r Max numbers of homes achieved. Please update/delete an existing one.'));
+        return();
+    );
+    if(global_hdata:uuid==null,global_hdata:uuid={});
+    global_names = keys(global_hdata:uuid);
+    if(global_names~name!=null, print(player,format(str('r %s is already in use. Therefore, updating the old home to new location. \n',name))));
+
+    global_hdata:uuid:name={};
+    global_hdata:uuid:name:'location' = pos(player);
+    global_hdata:uuid:name:'pitch' = player~'pitch';
+    global_hdata:uuid:name:'yaw' = player~'yaw';
+    global_hdata:uuid:name:'dimension' = player~'dimension';
+    _saveHData();

```

Comment:
Same as with another pr u've done I think, I think you should do saveData in __on_close() and that's it to avoid re-writing to disk a bunch of times

Reply:
Yh imma change this